common/db.py

```diff
@@ -619,26 +619,27 @@ def reset_not_finalized_batches_timestamps(self, app_name: str) -> None:
             batch["timestamp"] = now
 
     def reinitialize(
-        self,
-        app_name: str,
-        new_sequencer_id: str,
-        all_nodes_last_finalized_batch: Batch,
+            self,
+            app_name: str,
+            new_sequencer_id: str,
+            all_nodes_last_finalized_batch: Batch,
     ) -> None:
         """Reinitialize the database after a switch in the sequencer."""
         # TODO: Should get the batches from other nodes if they are missing.
         self.finalize_batches(
             app_name,
             signature_data={
-                "index": all_nodes_last_finalized_batch["index"],
-                "chaining_hash": all_nodes_last_finalized_batch["chaining_hash"],
-                "hash": all_nodes_last_finalized_batch["hash"],
-                "signature": all_nodes_last_finalized_batch["finalization_signature"],
-                "nonsigners": all_nodes_last_finalized_batch["finalized_nonsigners"],
-                "tag": all_nodes_last_finalized_batch["finalized_tag"],
+                "index": all_nodes_last_finalized_batch.get("index", 0),
```

Comment:
are these required when you're checking `all_nodes_last_finalized_batch` to not be empty before calling `reinitialize`?